cmd/collector/main.go

```diff
@@ -95,6 +96,12 @@ func main() {
 			c.Start(collectorOpts)
 
 			svc.RunAndThen(func() {
+				if closer, ok := spanWriter.(io.Closer); ok {
+					err := closer.Close()
+					if err != nil {
+						logger.Error("failed to close span writer", zap.Error(err))
+					}
+				}
```

Comment:
what about all-in-one, should it have similar code?

Reply:
It does already, which is the cause of the double-close:
https://github.com/jaegertracing/jaeger/blob/74856efebf703bd98742b96db48724dca4e9e7fb/cmd/all-in-one/main.go#L155-L160